django/db/models/base.py

```diff
@@ -2098,7 +2098,12 @@ def _check_constraints(cls, databases):
                         field = cls._meta.pk
                     else:
                         field = cls._meta.get_field(field_name)
-                    if not field.is_relation or field.many_to_many or field.one_to_many:
+                    if (
+                        not field.is_relation or
+                        field.many_to_many or
+                        field.one_to_many or
+                        field.one_to_one
+                    ):
```

Comment:
OK we need to revert to the previous solution, sorry. It will no longer raise an exception for joins on `OneToOneFields`, e.g.
```diff
diff --git a/tests/invalid_models_tests/test_models.py b/tests/invalid_models_tests/test_models.py
index 2802ce26e0..7a6725c52a 100644
--- a/tests/invalid_models_tests/test_models.py
+++ b/tests/invalid_models_tests/test_models.py
@@ -1796,6 +1796,7 @@ class ConstraintsTests(TestCase):
             field2 = models.PositiveSmallIntegerField()
             field3 = models.PositiveSmallIntegerField()
             parent = models.ForeignKey('self', models.CASCADE)
+            previous = models.OneToOneField('self', models.CASCADE)
 
             class Meta:
                 constraints = [
@@ -1810,9 +1811,12 @@ class ConstraintsTests(TestCase):
                     models.CheckConstraint(
                         name='name3', check=models.Q(parent__field3=models.F('field1'))
                     ),
+                    models.CheckConstraint(
+                        name='name4', check=models.Q(name=Lower('previous__name'))
+                    ),
                 ]
 
-        joined_fields = ['parent__field1', 'parent__field2', 'parent__field3', 'parent__name']
+        joined_fields = ['parent__field1', 'parent__field2', 'parent__field3', 'parent__name', 'previous__name']
         errors = Model.check(databases=self.databases)
         expected_errors = [
             Error(
```

Can you add this test in a separate commit?